### PR TITLE
1549 add relic viewer

### LINKED ISSUE #1549: Add relic viewer UI
- [x] Add `RelicCard` in `libs/sr/ui`, it just needs to display the stats of the Relic
  - Van did this already
- [ ] Add `RelicInventory` component in `libs/sr/ui` that renders all Relics in DB using `RelicCard` in a Grid
- [ ] Add some pagination to above (look at GO code for inspiration)
- [ ] Figure out a place to put this (Van will decide/help with this)
  - [ ] For testing, we can just put it on the main page

### PR DESCRIPTION
**Pull request by EuEric**
## Describe your changes

Created RelicInventory to display relics on a grid using the RelicCard. The in-house infinite scroll has been used for this.

## Issue or discord link

Resolves #1549

## Testing/validation

image

## Checklist before requesting a review (leave this PR as draft if any part of this list is not done.)

- [x] I have commented my code in hard-to understand areas.
- [ ] I have made corresponding changes to README or wiki.
- [ ] For front-end changes, I have updated the corresponding English translations.
- [x] I have run `yarn run mini-ci` locally to validate format and lint.
- [x] If I have added a new library or app, I have updated the deployment scripts to ignore changes as needed

### CODE REVIEW

File: apps/sr-frontend/src/app/App.tsx [hunk 1]
```diff
@@ -28,6 +29,7 @@ export default function App() {
               <Stack gap={1} pt={1}>
                 <CharacterSelector />
                 <Character />
+                <RelicInventory />
```
Comment (nguyentvan7) [inline apps/sr-frontend/src/app/App.tsx:32]: Wrap this in an Accordion (see stuff in Character)

Comment (nguyentvan7) [reply apps/sr-frontend/src/app/App.tsx:32]: I don't want to wrap the inner contents, so we can more easily transplant it into a separate page later down the line

File: libs/sr/ui/src/Relic/RelicInventory.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    () => relicsIds.slice(0, numShow),
+    [relicsIds, numShow]
+  )
+  const showingTextProps = {
+    numShowing: relicsIdsToShow.length,
+    totalShowing: totalShowing,
+  }
+
+  return (
+    <Accordion>
```
Comment (nguyentvan7) [inline libs/sr/ui/src/Relic/RelicInventory.tsx:49]: See other comment about moving Accordion placement

Comment (nguyentvan7) [inline libs/sr/ui/src/Relic/RelicInventory.tsx:49]: Also this accordion is full width for some reason
image

**Review by nguyentvan7 — CHANGES_REQUESTED** [review]

File: apps/sr-frontend/src/app/App.tsx (hunk 1 above)
Comment (EuEric) [reply apps/sr-frontend/src/app/App.tsx]: fixed on ba2186105d0f9a6fc46112936bd0b5dda574e2c7

File: libs/sr/ui/src/Relic/RelicInventory.tsx (hunk 2 above)
Comment (EuEric) [reply libs/sr/ui/src/Relic/RelicInventory.tsx]: fixed on ba2186105d0f9a6fc46112936bd0b5dda574e2c7

Comment (EuEric) [reply libs/sr/ui/src/Relic/RelicInventory.tsx]: fixed on ba2186105d0f9a6fc46112936bd0b5dda574e2c7

**Comment by frzyc** [comment]
If you want to link your PR to the issue, edit your description to:
```
Issue or discord link
- Resolves #1549
```

File: libs/sr/ui/src/Relic/RelicInventory.tsx (hunk 2 above)
Comment (EuEric) [reply libs/sr/ui/src/Relic/RelicInventory.tsx]: Any suggestion on how to fix this issue? @nguyentvan7

File: apps/sr-frontend/src/app/App.tsx [hunk 3]
```diff
@@ -28,6 +33,14 @@ export default function App() {
               <Stack gap={1} pt={1}>
                 <CharacterSelector />
                 <Character />
+                <Accordion>
```
Comment (frzyc) [inline apps/sr-frontend/src/app/App.tsx:36]: Wrap this `<Accordion>` component inside a `<Container>` component to make it not full-width.

**Review by frzyc — CHANGES_REQUESTED** [review]

File: libs/sr/ui/src/Relic/RelicInventory.tsx (hunk 2 above)
Comment (EuEric) [reply libs/sr/ui/src/Relic/RelicInventory.tsx]: fixed on 3a3a79cfa8aecc6d8a03fffe8ef070ab99b731a3

File: apps/sr-frontend/src/app/App.tsx (hunk 3 above)
Comment (EuEric) [reply apps/sr-frontend/src/app/App.tsx]: fixed on 3a3a79cfa8aecc6d8a03fffe8ef070ab99b731a3

**Review by nguyentvan7 — APPROVED** [review]

**Review by frzyc — APPROVED** [review]

**Comment by frzyc** [comment]
nit: wrap the component in a `<CardThemed>` to make it not blend in with background

File: apps/sr-frontend/src/app/App.tsx [hunk 4]
```diff
@@ -28,6 +34,16 @@ export default function App() {
               <Stack gap={1} pt={1}>
                 <CharacterSelector />
                 <Character />
+                <Container>
```
Comment (frzyc) [inline apps/sr-frontend/src/app/App.tsx:37]: @EuEric Can you wrap the Accordion in a `<CardThemed/>`, and move this entire thing as a separate component in `/RelicInventory.tsx`?

I would also recommend moving this Relic Inventory element below the `<RelicEditor/>` for slightly better flow.

**Review by frzyc — CHANGES_REQUESTED** [review]
Some more cleanup

File: apps/sr-frontend/src/app/App.tsx (hunk 4 above)
Comment (EuEric) [reply apps/sr-frontend/src/app/App.tsx:37]: Sure, no problem, will work on it starting next week

Comment (EuEric) [reply apps/sr-frontend/src/app/App.tsx]: Moved whole Accordion wrapped in CardThemed inside RelivInventory and placed RelicEditor ahead

File: libs/sr/ui/src/Relic/RelicInventory.tsx [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+    [relicsIds, numShow]
+  )
+  const showingTextProps = {
+    numShowing: relicsIdsToShow.length,
+    totalShowing: totalShowing,
+  }
+
+  return (
+    <Container>
+      <CardThemed>
```
Comment (nguyentvan7) [inline libs/sr/ui/src/Relic/RelicInventory.tsx:51]:
```suggestion
      <CardThemed bgt='dark'>
```

Comment (EuEric) [reply libs/sr/ui/src/Relic/RelicInventory.tsx:51]: changed in last commit

**Comment by nguyentvan7** [comment]
format issues

**Comment by nguyentvan7** [comment]
oops and they were my fault :^)

**Review by nguyentvan7 — APPROVED** [review]

**Review by frzyc — APPROVED** [review]

**Pull request merged**